packages/button/addon/components/nucleus-button.js

```diff
@@ -419,6 +420,15 @@ class NucleusButton extends Component {
   */
   click() {
     let action = this.get('onClick');
+    if(this.type === "toggle") {
```

Comment:
This block has to come after 'action' check. Also, kindly verify how this behaviour affects async button functionality.